src/Middleware/CORS/src/Infrastructure/ICorsAttribute.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Cors.Infrastructure
+{
+    /// <summary>
+    /// A marker interface which can be used to identify CORS metdata.
+    /// </summary>
+    public interface ICorsAttribute
```

Comment:
What about `ICorsMetadata`? Putting attribute in the name seems silly since it's not an attritube.

Reply:
Already done locally.

Will look like:

* ICorsMetadata
  * IEnableCorsAttribute
  * IDisableCorsAttribute
  * ICorsPolicyMetadata

ICorsPolicyMetadata would have the policy name and a policy on it to support https://github.com/aspnet/AspNetCore/pull/4460#discussion_r239676450. Constructor would only allow one.

Thoughts? An alternate is ICorsPolicyMetadata only have the CorsPolicy on it and we leave IEnableCorsAttribute as the way to set the policy name.